app/src/main/java/com/mk/tasky/agenda/detail/reminder/presentation/DetailReminderViewModel.kt

```diff
@@ -36,6 +40,19 @@ class DetailReminderViewModel @Inject constructor(
                 state = state.copy(
                     isEditing = false
                 )
+                viewModelScope.launch {
+                    saveReminder(
+                        id = state.id,
+                        title = state.title,
+                        description = state.description,
+                        time = state.time,
+                        date = state.date,
+                        reminder = state.reminder
+                    )
+                }
+                state = state.copy(
+                    shouldExit = true
```

Comment:
What if exiting is faster here than saving the reminder? That would clear the VM, cancel all coroutines in viewModelScope and therefore cancel saving the reminder.

The solution would be using `NonCancellable` or a scope that outlives the VM:
```
viewModelScope.launch {
    withContext(NonCancellable) {
         // save reminder
    }
}
```

Reply:
Oh wow you are right
Thanks for the tip!